libs/ColorKit/tests/ColorKit_test_gradient.cpp

```diff
@@ -0,0 +1,109 @@
+// Leka - LekaOS
+// Copyright 2022 APF France handicap
+// SPDX-License-Identifier: Apache-2.0
+
+#include <array>
+#include <memory>
+
+#include "ColorKit.h"
+#include "CoreLED.h"
+#include "CoreSPI.h"
+#include "gtest/gtest.h"
+
+using namespace leka;
+
+class ColorKitGradientTest : public ::testing::Test
+{
+  protected:
+	ColorKitGradientTest() = default;
+
+	// void SetUp() override {}
+	// void TearDown() override {}
+
+	static constexpr int number_of_leds = 20;
+
+	std::array<RGB, number_of_leds> expected_colors {};
+
+	CoreSPI spi {NC, NC, NC, NC};
+	CoreLED<number_of_leds> leds {spi};
+};
+
+TEST_F(ColorKitGradientTest, gradientWhiteToBlack)
+{
+	auto expected_colors = std::to_array<RGB>(
+		{RGB::white,		  RGB {217, 217, 217}, RGB {183, 183, 183}, RGB {152, 152, 152}, RGB {125, 125, 125},
+		 RGB {102, 102, 102}, RGB {82, 82, 82},	   RGB {64, 64, 64},	RGB {49, 49, 49},	 RGB {37, 37, 37},
+		 RGB {27, 27, 27},	  RGB {19, 19, 19},	   RGB {13, 13, 13},	RGB {8, 8, 8},		 RGB {5, 5, 5},
+		 RGB {2, 2, 2},		  RGB {1, 1, 1},	   RGB {0, 0, 0},		RGB {0, 0, 0},		 RGB::black});
+
+	for (auto range = 0; range < number_of_leds; range++) {
+		RGB col_rgb = ColorKit::colorGradient(RGB::white, RGB::black, range);
+		leds.setColorAtIndex(range, col_rgb);
+	}
+
+	EXPECT_TRUE(std::equal(expected_colors.begin(), expected_colors.end(), leds.getColor().begin()));
+}
+
+TEST_F(ColorKitGradientTest, gradientWhiteToPureRed)
+{
+	auto expected_colors = std::to_array<RGB>(
+		{RGB::white,		  RGB {255, 234, 229}, RGB {255, 213, 205}, RGB {255, 194, 183}, RGB {255, 176, 162},
+		 RGB {255, 158, 142}, RGB {255, 142, 124}, RGB {255, 126, 108}, RGB {255, 111, 93},	 RGB {255, 98, 79},
+		 RGB {255, 84, 66},	  RGB {255, 72, 55},   RGB {255, 61, 45},	RGB {255, 50, 35},	 RGB {255, 40, 27},
+		 RGB {255, 30, 20},	  RGB {255, 22, 14},   RGB {255, 14, 8},	RGB {255, 7, 4},	 RGB::pure_red});
+
+	for (auto range = 0; range < number_of_leds; range++) {
+		RGB col_rgb = ColorKit::colorGradient(RGB::white, RGB::pure_red, range);
+		leds.setColorAtIndex(range, col_rgb);
+	}
+
+	EXPECT_TRUE(std::equal(expected_colors.begin(), expected_colors.end(), leds.getColor().begin()));
+}
+
+TEST_F(ColorKitGradientTest, gradientWhiteToPureGreen)
+{
+	auto expected_colors = std::to_array<RGB>(
+		{RGB::white,		  RGB {237, 255, 235}, RGB {220, 255, 216}, RGB {203, 255, 198}, RGB {187, 255, 180},
+		 RGB {172, 255, 164}, RGB {156, 255, 148}, RGB {142, 255, 132}, RGB {127, 255, 118}, RGB {114, 255, 104},
+		 RGB {100, 255, 91},  RGB {87, 255, 78},   RGB {75, 255, 66},	RGB {63, 255, 55},	 RGB {52, 255, 44},
+		 RGB {40, 255, 34},	  RGB {30, 255, 25},   RGB {19, 255, 16},	RGB {9, 255, 8},	 RGB::pure_green});
+
+	for (auto range = 0; range < number_of_leds; range++) {
+		RGB col_rgb = ColorKit::colorGradient(RGB::white, RGB::pure_green, range);
+		leds.setColorAtIndex(range, col_rgb);
+	}
+
+	EXPECT_TRUE(std::equal(expected_colors.begin(), expected_colors.end(), leds.getColor().begin()));
```

Comment:
Plus simplement (et sans utiliser les LEDs)
```suggestion
	std::array<RGB, number_of_leds> actual_colors = {};

	for (auto range = 0; range < number_of_leds; range++) {
		RGB col_rgb				= ColorKit::colorGradient(RGB::white, RGB::pure_green, range);
		actual_colors.at(range) = col_rgb;
	}

	EXPECT_TRUE(std::equal(expected_colors.begin(), expected_colors.end(), actual_colors.begin()));
```